### PR TITLE
feature: create ProcessingJob from ARN and from name

### PR DESCRIPTION
**Pull request by ajaykarpur**
*Description of changes:*
This feature allows the user to create a ProcessingJob instance either using a Processing job ARN or using a Processing job name.

*Testing done:*
- Added a unit test that covers both `from_processing_arn` and `from_processing_name`

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have used the regional endpoint when creating S3 and/or STS clients (if appropriate)
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/model_monitor/model_monitoring.py [hunk 1]
```diff
@@ -1761,7 +1761,7 @@ def _get_default_image_uri(region):
 class BaseliningJob(ProcessingJob):
     """Provides functionality to retrieve baseline-specific files output from baselining job."""
 
-    def __init__(self, sagemaker_session, job_name, inputs, outputs):
+    def __init__(self, sagemaker_session, job_name, inputs, outputs, output_kms_key):
```
Comment (laurenyu) [inline src/sagemaker/model_monitor/model_monitoring.py]: do we need to default this to `None`? (are we worried about this being a breaking change?)

File: tests/unit/test_processing.py [hunk 2]
```diff
@@ -388,6 +437,14 @@ def test_processor_with_all_parameters(sagemaker_session):
     sagemaker_session.process.assert_called_with(**expected_args)
 
 
+def test_processing_job_from_processing_arn(sagemaker_session):
+    processing_job = ProcessingJob.from_processing_arn(
+        sagemaker_session=sagemaker_session,
+        processing_job_arn="arn:aws:sagemaker:dummy-region:dummy-account-number:processing-job/dummy-job-name",
+    )
+    assert isinstance(processing_job, ProcessingJob)
```
Comment (laurenyu) [inline tests/unit/test_processing.py]: should this also check that attributes are set correctly?

File: src/sagemaker/model_monitor/model_monitoring.py (hunk 1 above)
Comment (ajaykarpur) [reply src/sagemaker/model_monitor/model_monitoring.py]: Good call-- defaulted this to `None`.

File: tests/unit/test_processing.py (hunk 2 above)
Comment (ajaykarpur) [reply tests/unit/test_processing.py]: Added more checks for the attributes.

File: src/sagemaker/model_monitor/model_monitoring.py [hunk 3]
```diff
@@ -1773,12 +1773,17 @@ def __init__(self, sagemaker_session, job_name, inputs, outputs):
             job_name (str): Name of the Amazon SageMaker Model Monitoring Baselining Job.
             inputs ([sagemaker.processing.ProcessingInput]): A list of ProcessingInput objects.
             outputs ([sagemaker.processing.ProcessingOutput]): A list of ProcessingOutput objects.
+            output_kms_key (str): The output kms key associated with the job.
```
Comment (laurenyu) [inline src/sagemaker/model_monitor/model_monitoring.py]: note the default here and what happens if you do/don't set it

File: src/sagemaker/processing.py [hunk 4]
```diff
@@ -502,9 +502,10 @@ def _set_entrypoint(self, command, user_script_name):
 class ProcessingJob(_Job):
     """Provides functionality to start, describe, and stop processing jobs."""
 
-    def __init__(self, sagemaker_session, job_name, inputs, outputs):
+    def __init__(self, sagemaker_session, job_name, inputs, outputs, output_kms_key):
```
Comment (laurenyu) [inline src/sagemaker/processing.py]: same concern here re: set a default

**Pull request merged**